src/common/controllers/ads-controller.js

```diff
@@ -139,6 +141,12 @@ class AdsController extends FakeEventTarget implements IAdsController {
     this._eventManager.listen(this._player, CustomEventType.PLAYER_DESTROY, () => this._destroy());
     this._eventManager.listenOnce(this._player, Html5EventType.ENDED, () => this._onEnded());
     this._eventManager.listenOnce(this._player, CustomEventType.PLAYBACK_ENDED, () => this._onPlaybackEnded());
+    this._eventManager.listen(this._player, AdEventType.AD_RESUMED, () => {
+      this._isAdPlaying = true;
+    });
+    this._eventManager.listen(this._player, AdEventType.AD_PAUSED, () => {
+      this._isAdPlaying = false;
```

Comment:
also on `AD_ERROR` and `AD_ENDED`

Reply:
why `AD_ENDED`?

Reply:
We have the `AD_BREAK_END` which cleans the _adBreak property. Isn't it enough?

Reply:
added in critical error

Reply:
the Q is why not `AD_ENDED`? once the ad is ended it means is not playing

Reply:
I mean `AD_COMPLETED` sorry

Reply:
AD_ENDED will be the same as AD_BREAK_END for one ad and for multiple I don't think you want to change it false and immediately to true, don't you think?

Reply:
@yairans we should split the playback playing or ad break playing, you want to know which content to pause, so `AD_COMPLETED` will change it after every ad on ad break

Reply:
first I dont see `_adIsPlaying` becoming `false` on `AD_BREAK_START`
also, if it's becoming `true` on `AD_STARTED` to be symmetrically it's need to `false` on `AD_COMPLETED`

Reply:
The isAdPlaying returns _adIsPlaying && isAdBreak so it becomes false if you're not in adBreak, for `AD_STARTED` I agree it could change to `AD_BREAK_START`, Roy wanted to use the same logic as we have on UI

Reply:
the api is correct but why to keep incorrect flag? tomorrow someone will rely on this flag and run to mistake.
If the ui has mistakes there is no reason to keep them in other repos

Reply:
if you want to use `AD_BREAk_START` and `AD_BREAK_END` I dont care

Reply:
I added on AD_COMPLETED.

Reply:
BTW I don't think it's if we want to or not we have to use `AD_BREAK_START` and `AD_BREAK_END` that when we pause and play the playback there's no reason that it won't be consistent